### PR TITLE
Allows migrations to be optional

### PR DESCRIPTION
**Pull request by bahelms**
Introduces a new action input `MIGRATIONS` to be able to skip migrations. It defaults to `true`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ deploy:
         GIGALIXIR_PASSWORD: ${{ secrets.GIGALIXIR_PASSWORD }}
         GIGALIXIR_APP: my-gigalixir-app # Feel free to also put this in your secrets
         SSH_PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
+        RUN_MIGRATIONS: false  # defaults to "true"
```
Comment (mhanberg) [inline README.md]:
```suggestion
       MIGRATIONS: false  # defaults to true
```
Lets call this `MIGRATIONS` and make the default a boolean

File: action.yml [hunk 2]
```diff
@@ -13,6 +13,10 @@ inputs:
   GIGALIXIR_APP:
     description: 'Your gigalixir app name'
     required: true
+  RUN_MIGRATIONS:
+    description: 'Whether or not to run migrations'
+    required: true
+    default: 'true'
```
Comment (mhanberg) [inline action.yml]:
```suggestion
  MIGRATIONS:
    description: 'Configuration for migrations'
    required: true
    default: true
```

File: index.js [hunk 3]
```diff
@@ -70,11 +70,12 @@ async function getCurrentRelease(app) {
 }
 
 async function run() {
-  try { 
+  try {
     const gigalixirUsername = core.getInput('GIGALIXIR_USERNAME', { required: true });
     const gigalixirPassword = core.getInput('GIGALIXIR_PASSWORD', { required: true });
     const sshPrivateKey = core.getInput('SSH_PRIVATE_KEY', { required: true });
     const gigalixirApp = core.getInput('GIGALIXIR_APP', { required: true });
+    const runMigrations = core.getInput('RUN_MIGRATIONS', { required: true });
```
Comment (mhanberg) [inline index.js]:
```suggestion
    const migrations = core.getInput('MIGRATIONS', { required: true });
```

File: index.js [hunk 4]
```diff
@@ -105,19 +106,21 @@ async function run() {
...
-        await exec.exec(`gigalixir ps:migrate -a ${gigalixirApp}`)
-      });
-    } catch (error) {
-      core.warning(`Migration failed, rolling back to the previous release: ${currentRelease}`);
-      await core.group("Rolling back", async () => {
-        await exec.exec(`gigalixir releases:rollback -a ${gigalixirApp}`)
-      });
-
-      core.setFailed(error.message);
+    if (runMigrations === "true") {
```
Comment (mhanberg) [inline index.js]:
```suggestion
    if (migrations) {
```

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "gigalixir-action",
-  "version": "0.1.0",
+  "version": "0.2.0",
```
Comment (mhanberg) [inline package.json]: Can you revert the version change? I will do this myself when i cut the next release.

File: README.md (hunk 1 above)
Comment (bahelms) [reply README.md]: Sounds good. I assumed the input would come through as a string; I wasn't sure if would be converted.

File: package.json (hunk 5 above)
Comment (bahelms) [reply package.json]: 👍

**Comment by bahelms** [comment]
Bump. I have a Phoenix app on Gigalixir that doesn't use a database, and I was hoping to use this Github Action instead of rolling my own. I'd appreciate a re-review. Thank you!

**Comment by mhanberg** [comment]
@bahelms Could you test out this new functionality before I tag the next release? You should be able to switch the `@v0.1.0` in your workflow with `@master`.

Thanks!

**Comment by bahelms** [comment]
After trying it out, I've discovered that `const migrations = core.getInput('MIGRATIONS', { required: true });` returns a string, so if `MIGRATIONS` is `false` it comes out as `"false"` and is always truthy.  So the condition should probably be `migrations !== 'true'` due to the default. https://github.com/actions/toolkit/blob/master/packages/core/src/core.ts#L67.

**Comment by mhanberg** [comment]
Oh that is annoying, sorry about that. 

Can you open a new PR? If not I can fix it on master.

**Comment by bahelms** [comment]
No worries. I'll do a new PR to fix it.

**Pull request merged**